build.sbt

```diff
@@ -1,47 +1,87 @@
-  // the deprecation:false flag is only supported by scala >= 2.11.3, but needed for scala >= 2.11.0 to avoid warnings
-  CrossVersion.partialVersion(scalaVersion.value) match {
-    case Some((2, scalaMinor)) if scalaMinor >= 11 =>
-      // For scala versions >= 2.11.3
-      Seq("-Xfatal-warnings", "-deprecation:false")
-    case Some((2, scalaMinor)) if scalaMinor < 11 =>
-      // For scala versions 2.10.x
-      Seq("-Xfatal-warnings", "-deprecation")
-  }
-} ++ Seq(
-  "-feature",
-  "-Ywarn-dead-code",
-  "-encoding", "UTF-8"
-)
+  libraryDependencies ++= Seq(
+    "joda-time" % "joda-time" % "2.8",
+    "org.joda" % "joda-convert" % "1.7",
+    "org.scalacheck" %% "scalacheck" % "1.11.3" % "test",
+    "org.scalatest" %% "scalatest" % "2.2.4" % "test"
+  ),
 
-libraryDependencies ++= Seq(
-  "org.mongodb" %% "casbah-query" % "2.8.0",
-  "org.mongodb" %% "casbah-core" % "2.8.0",
-  "org.mongodb" % "mongo-java-driver" % "2.13.1",
-  "org.scalacheck" %% "scalacheck" % "1.11.3",
-  "org.scalatest" %% "scalatest" % "2.2.4" % "test",
-  "com.typesafe.play" %% "play-functional" % "2.4.0-M2"
-)
+  // disable compilation of ScalaDocs, since this always breaks on links
+  sources in(Compile, doc) := Seq.empty,
 
-// disable compilation of ScalaDocs, since this always breaks on links
-sources in(Compile, doc) := Seq.empty
+  // disable publishing empty ScalaDocs
+  publishArtifact in (Compile, packageDoc) := false,
+
+  licenses += ("Apache-2.0", url("http://opensource.org/licenses/apache-2.0"))
+
+) ++ bintraySettings ++ bintrayPublishSettings
+
+lazy val `bson-adt-core` = (project in file("bson-adt-core")).settings(common).settings(
+  name := "bson-adt-core",
+  libraryDependencies ++= Seq(
+    "com.typesafe.play" %% "play-functional" % "2.4.0-M2",
+    "org.mongodb" % "bson" % "3.0.0"
+  )
+)
 
-// disable publishing empty ScalaDocs
-publishArtifact in (Compile, packageDoc) := false
+lazy val `bson-adt-mongo2` = (project in file("bson-adt-mongo2")).settings(common).settings(
+  name := "bson-adt-mongo2",
+  libraryDependencies ++= Seq(
+    "com.typesafe.play" %% "play-functional" % "2.4.0-M2",
+    "org.mongodb" % "mongo-java-driver" % "2.13.1"
+  )
+).dependsOn(
+    `bson-adt-core` % "compile->compile;test->test"
+  )
 
-bintraySettings ++ bintrayPublishSettings
+lazy val `bson-adt-casbah` = (project in file("bson-adt-casbah")).settings(common).settings(
+  name := "bson-adt-casbah",
+  libraryDependencies ++= Seq(
+    "org.mongodb" %% "casbah-core" % "2.8.0",
+    "org.mongodb" %% "casbah-query" % "2.8.0"
+  )
+).dependsOn(
+    `bson-adt-mongo2` % "compile->compile;test->test"
+  )
 
-licenses += ("Apache-2.0", url("http://opensource.org/licenses/apache-2.0"))
+lazy val `bson-adt-mongo` = (project in file("bson-adt-mongo3")).settings(common).settings(
+  name := "bson-adt-mongo",
+  libraryDependencies ++= Seq(
+    "org.mongodb" % "mongo-java-driver" % "3.0.2"
+  )
+).dependsOn(
+    `bson-adt-core` % "compile->compile;test->test"
+  )
```

Comment:
Git is all confused because I made this a multi-project build.

Basically `bson-adt` split into:
1. `bson-adt-core` with only a dependence on `org.mongodb:bson:3.0.0`
2. `bson-adt-mongo2` with dependence on `bson-adt-core` and `org.mongodb:mongo-java-driver:2.13.1`
3. `bson-adt-casbah` with a dependence on `bson-adt-mongo2` and casbah
4. `bson-adt-mongo` with a dependence on `bson-adt-core` and `org.mongodb:mongo-java-driver:3.0.2`

I figure this is alright since `org.mongodb` split out the `bson` library in v3, so in all future versions, `bson-adt-mongo` would be the hub of any new extensions (include mongodb drivers themselves), but the bson version could remain the lowest necessary.